airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -73,191 +73,129 @@
-                maybeJobId.get(),
-                maybeAttemptId.get(),
-                FailureHelper.failureSummaryForCancellation(maybeJobId.get(), maybeAttemptId.get(), failures, partialSuccess)));
-        resetNewConnectionInput(connectionUpdaterInput);
-      } else if (workflowState.isFailed()) {
-        reportFailure(connectionUpdaterInput);
-      } else {
-        // report success
-        reportSuccess(connectionUpdaterInput);
+      if (workflowState.isCancelled()) {
+        reportCancelledAndContinueWith(false, connectionUpdaterInput);
       }
-      continueAsNew(connectionUpdaterInput);
+
     } catch (final Exception e) {
       log.error("The connection update workflow has failed, will create a new attempt.", e);
-
-      reportFailure(connectionUpdaterInput);
-      continueAsNew(connectionUpdaterInput);
+      reportFailure(connectionUpdaterInput, null);
+      prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
     }
   }
 
-  private void reportSuccess(final ConnectionUpdaterInput connectionUpdaterInput) {
+  private CancellationScope generateSyncWorkflowRunnable(ConnectionUpdaterInput connectionUpdaterInput) {
+    return Workflow.newCancellationScope(() -> {
+      connectionId = connectionUpdaterInput.getConnectionId();
+
+      if (connectionUpdaterInput.getWorkflowState() != null) {
+        workflowState = connectionUpdaterInput.getWorkflowState();
+      }
+
+      workflowState.setResetConnection(connectionUpdaterInput.isResetConnection());
+
+      Duration timeToWait = getTimeToWait(connectionUpdaterInput.getConnectionId());
+
+      Workflow.await(timeToWait,
+          () -> skipScheduling() || connectionUpdaterInput.isFromFailure());
+
+      val needToProcessSync = !workflowState.isUpdated() && !workflowState.isDeleted();
+
+      if (needToProcessSync) {
+        jobId = getOrCreateJobId(connectionUpdaterInput);
+
+        attemptId = createAttemptId(jobId);
+
+        final GeneratedJobInput jobInputs = getJobInput();
+
+        reportJobStarting();
+        StandardSyncOutput standardSyncOutput = null;
+        try {
+          standardSyncOutput = runChildWorkflow(jobInputs);
+
+          workflowState.setFailed(getFailStatus(standardSyncOutput));
+
+          if (workflowState.isFailed()) {
+            reportFailure(connectionUpdaterInput, standardSyncOutput);
+            prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
+          } else {
+            reportSuccess(connectionUpdaterInput, standardSyncOutput);
+            prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
+          }
+        } catch (final ChildWorkflowFailure childWorkflowFailure) {
+          if (childWorkflowFailure.getCause() instanceof CanceledFailure) {
+            // do nothing, cancellation handled by cancellationScope
+
+          } else if (childWorkflowFailure.getCause() instanceof ActivityFailure) {
+            final ActivityFailure af = (ActivityFailure) childWorkflowFailure.getCause();
```

Comment:
```suggestion
          } else if (childWorkflowFailure.getCause() instanceof final ActivityFailure af) {
```

Reply:
we can take advantage of the above cast here to avoid casting again

Reply:
Done, I tried to use a switch but it is in preview only unfortunately.